web/modules/custom/joinup_core/joinup_core.install

```diff
@@ -60,3 +62,94 @@ function joinup_core_requirements($phase) {
 function joinup_core_update_8100(&$sandbox) {
   \Drupal::service('module_installer')->install(['config_sync']);
 }
+
+/**
+ * Update the type of collections, solutions, releases and distributions.
+ *
+ * Two things will happen here.
+ * The first is that all related field mappings will be updated.
+ * The second is that all entities in the database will be updated.
+ */
+function joinup_core_update_8102() {
+  $entity_type_id = 'rdf_entity';
+  $query = Database::getConnection('default', 'sparql_default');
+  $bundles = [
+    'collection' => [
+      'old_uri' => 'http://www.w3.org/ns/adms#AssetRepository',
+      'new_uri' => 'http://www.w3.org/ns/dcat#Catalog',
+    ],
+    'solution' => [
+      'old_uri' => 'http://www.w3.org/ns/adms#Asset',
+      'new_uri' => 'http://www.w3.org/ns/dcat#Dataset',
+    ],
+    'asset_release' => [
+      'old_uri' => 'http://www.w3.org/ns/adms#Asset',
+      'new_uri' => 'http://www.w3.org/ns/dcat#Dataset',
+    ],
+    'asset_distribution' => [
+      'old_uri' => 'http://www.w3.org/ns/adms#AssetDistribution',
+      'new_uri' => 'http://www.w3.org/ns/dcat#Distribution',
+    ],
+  ];
+
+  $bundle_predicate = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#type';
+  foreach ($bundles as $bundle => $properties) {
+    $rdf_entity_mapping = RdfEntityMapping::loadByName($entity_type_id, $bundle);
+
+    // Update the mappings.
+    $rdf_entity_mapping->setRdfType($properties['new_uri']);
```

Comment:
Nit: Just for fanciness, this could be written more compact with chaining `$rdf_entity_mapping->setRdfType($properties['new_uri'])->save();`.